### PR TITLE
Explore: Prometheus query errors now show

### LINKED ISSUE #17435: Explore: Prometheus query error not shown
**What happened**:

Query syntax error was not shown.

**How to reproduce it (as minimally and precisely as possible)**:

0. Explore with Prometheus
1. Write a query with a syntax error
2. Run query

Result: empty graph, no error shown

**Anything else we need to know?**:

Cortex replied to the query with a 400.
Here is the body of the response:
```
{"status":"error","errorType":"bad_data","error":"invalid parameter 'query': parse error at char 16: unknown function with name \"coun\""}
```

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Displays an error message when a query with invalid syntax is made.

**Which issue(s) this PR fixes**:
Fixes #17435

### CODE REVIEW

**Review by marefr — CHANGES_REQUESTED** [review]
The code looks good and works as expected :+1: However, I have no experience with rxjs so going to ask Hugo to have a look at this before merging.

Would like you to add at least one test of the datasource query function when an error is returned from the datasource. Should be able to use this test as a template for testing query that returns an error: https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/prometheus/specs/datasource.test.ts#L470

File: public/app/plugins/datasource/prometheus/datasource.ts [hunk 1]
```diff
@@ -399,16 +404,16 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     if (this.queryTimeout) {
       data['timeout'] = this.queryTimeout;
     }
-    return this._request(url, data, { requestId: query.requestId, headers: query.headers }).catch((err: any) =>
-      this.handleErrors(err, query)
-    );
+    return this._request(url, data, { requestId: query.requestId, headers: query.headers }).catch((err: any) => {
+      if (err.cancelled) {
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/prometheus/datasource.ts]: Small question, why is this if statement moved outside the handleErrors function?
```
if (err.cancelled) {
```

Comment (kaydelaney) [reply public/app/plugins/datasource/prometheus/datasource.ts]: I think my thinking was that I wanted it to match how things were done in https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/loki/datasource.ts#L124 but I can change it if that's not necessary

Comment (hugohaggmark) [reply public/app/plugins/datasource/prometheus/datasource.ts]: Ok, thanks!

**Review by hugohaggmark — APPROVED** [review]
LGTM!

**Review by marefr — APPROVED** [review]
Great :+1:

**Pull request merged**